### PR TITLE
[uma-voting] add new strategy uma-voting

### PR DESCRIPTION
**Pull request by abg4**
Signed-off-by: Alex Gaines <againes@umaproject.org>

Changes proposed in this pull request:
- Adds new uma-voting strategy

### CODE REVIEW

File: src/strategies/uma-voting/README.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# Contract call strategy
```
Comment (ChaituVR) [inline src/strategies/uma-voting/README.md:1]:
```suggestion
# uma-voting strategy
```

File: src/strategies/uma-voting/index.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  {
+    inputs: [{ internalType: 'address', name: '', type: 'address' }],
+    name: 'designatedVotingContracts',
+    outputs: [
+      { internalType: 'contract DesignatedVoting', name: '', type: 'address' }
+    ],
+    stateMutability: 'view',
+    type: 'function'
+  }
+];
```
Comment (ChaituVR) [inline src/strategies/uma-voting/index.ts:17]:
```suggestion
const designatedVotingContractAbi = [
  'function designatedVotingContracts(address) view returns (address)'
];
```

File: src/strategies/uma-voting/index.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    network,
+    provider,
+    [options.methodABI],
+    votingAddress.map((address: any) => [
+      options.address,
+      options.methodABI.name,
+      getArgs(options, address)
+    ]),
+    { blockTag }
+  );
```
Comment (ChaituVR) [inline src/strategies/uma-voting/index.ts:58]: You want a way to call different methods or will it be calling `balanceOf` method everytime? maybe you can call erc20balanceof strategy directly like here https://github.com/snapshot-labs/snapshot-strategies/blob/master/src/strategies/erc20-balance-of-delegation/index.ts#L24

**Review by ChaituVR — CHANGES_REQUESTED** [review]

File: src/strategies/uma-voting/index.ts (hunk 3 above)
Comment (abg4) [reply src/strategies/uma-voting/index.ts]: We will only ever be calling `balanceOf` method. Let me know if the update works but removed the abi from the readme and examples.json files.

**Review by ChaituVR — APPROVED** [review]

**Pull request merged**